src/routes/namespaces/[namespace]/workflows/[workflow]/[run]/history/__layout@root.svelte

```diff
@@ -0,0 +1,38 @@
+<script lang="ts">
+  import { onDestroy } from 'svelte';
+  import { page } from '$app/stores';
+  import { clearPreviousEventParameters } from '$lib/stores/previous-events';
+
+  import WorkflowRunLayout from '$lib/layouts/workflow-run-layout.svelte';
+  import WorkflowHistoryLayout from '$lib/layouts/workflow-history-layout.svelte';
+  import WorkflowHistoryFeed from '$lib/pages/workflow-history-feed.svelte';
+  import WorkflowHistoryJson from '$lib/pages/workflow-history-json.svelte';
+  import WorkflowHistoryCompact from '$lib/pages/workflow-history-compact.svelte';
+
+  import PageTitle from '$lib/components/page-title.svelte';
+
+  onDestroy(() => {
+    clearPreviousEventParameters();
+  });
+
+  const workflow = $page.params.workflow;
+
+  $: compactPage = $page.url.pathname.endsWith('compact');
+  $: jsonPage = $page.url.pathname.endsWith('json');
```

Comment:
Do you think we even need routes for these views? Maybe could just make the last selected view a sticky preference.

Reply:
We don't, I was being overly cautious about removing them since there will be old links with /feed or /compact. What I did in the latest commit is remove them from any internal routes, but if a user does go to .../history/feed or .../history/compact or .../history/json, i redirect them to /history

Reply:
So I'm using the existing sticky view preference (eventViewType) to render the right view.

```
  const views = {
    feed: WorkflowHistoryFeed,
    compact: WorkflowHistoryCompact,
    json: WorkflowHistoryJson,
  };
  $: view = views[$eventViewType] ?? WorkflowHistoryFeed;

 <svelte:component this={view} />

```

Reply:
Beautiful 🎨